pkg/querier/metadata_handler.go

```diff
@@ -56,8 +83,17 @@ func NewMetadataHandler(m MetadataSupplier) http.Handler {
 		// Put all the elements of the pseudo-set into a map of slices for marshalling.
 		metrics := map[string][]metricMetadata{}
 		for _, m := range resp {
+			if metric != "" && m.Metric != metric {
```

Comment:
Might be worth adding a comment explaining why we're enforcing these here and in the ingesters